From 7d993d072ebfba6fa607d9d91726ed3fb078883a Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Mon, 24 Jun 2019 12:58:49 +0530
Subject: [PATCH 309/386] octeontx2-af: Display message timeout error

Displaying an error if time out occurs while waiting
for response helps in debugging.

Change-Id: I9532501d1cc5e60e9364f94f20d79b585ea03204
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/11503
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/mbox.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/mbox.c b/drivers/net/ethernet/marvell/octeontx2/af/mbox.c
index 288e1918ee60..ee019ffa7274 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/mbox.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/mbox.c
@@ -135,13 +135,16 @@ EXPORT_SYMBOL(otx2_mbox_init);
 int otx2_mbox_wait_for_rsp(struct otx2_mbox *mbox, int devid)
 {
 	struct otx2_mbox_dev *mdev = &mbox->dev[devid];
+	struct device *sender = &mbox->pdev->dev;
 	int timeout = 0, sleep = 1;
 
 	while (mdev->num_msgs != mdev->msgs_acked) {
 		msleep(sleep);
 		timeout += sleep;
-		if (timeout >= MBOX_RSP_TIMEOUT)
+		if (timeout >= MBOX_RSP_TIMEOUT) {
+			dev_err(sender, "timed out while waiting for rsp\n");
 			return -EIO;
+		}
 	}
 	return 0;
 }
-- 
2.17.1

